documentation/design-docs/dotnet-monitor.yaml

```diff
@@ -0,0 +1,473 @@
+        description: "Process ID of the process"
+        required: true
+        type: "string"
+      responses:
+        200:
+          description: "Success"
+          schema:
+            format: "byte"
+        400:
+          description: "Bad request"
+          schema:
+            $ref: "#/definitions/Error"
+    post:
+      tags:
+      - "logs"
+      operationId: "postlogs"
+      produces:
+      - "application/x-ndjson"
+      parameters:
+      - name: "uid"
+        in: "path"
+        description: "Process ID of the process"
+        required: true
+        type: "string"
+      - name: "configuration"
+        in: "body"
+        schema:
+          type: "array"
+          items:
+            $ref: "#/definitions/LogFilter"
+      responses:
+        200:
+          description: "Success"
+          schema:
+            format: "byte"
+        400:
+          description: "Bad request"
+          schema:
+            $ref: "#/definitions/Error"
+            
+  /metrics:
+    get:
+      tags:
+      - "metrics"
+      operationId: "getdefaultmetrics"
+      produces:
+      - "text/plain; version=0.0.4"
+      - "application/json"
+      responses:
+        200:
+          description: "Success"
+          schema:
+           $ref: "#/definitions/Metrics" 
+        400:
+          description: "Bad request"
+          schema:
+            $ref: "#/definitions/Error"
+  /metrics/{uid}:
+    get:
+      tags:
+      - "metrics"
+      operationId: "getmetrics"
+      produces:
+      - "text/plain; version=0.0.4"
+      - "application/json"
+      parameters:
+      - name: "uid"
+        in: "path"
+        description: "Process ID of the process"
+        required: true
+        type: "string"
+      responses:
+        200:
+          description: "Success"
+          schema:
+           $ref: "#/definitions/Metrics" 
+        400:
+          description: "Bad request"
+          schema:
+            $ref: "#/definitions/Error"
+    post:
+      tags:
+      - "metrics"
+      operationId: "postmetrics"
+      produces:
+      - "text/plain; version=0.0.4"
+      parameters:
+      - name: "uid"
+        in: "path"
+        description: "Process ID of the process"
+        required: true
+        type: "string"
+      - name: "configuration"
+        in: "body"
+        schema:
+          type: "array"
+          items:
+            $ref: "#/definitions/EventPipeProvider"
+      responses:
+        200:
+          description: "Success"
+          schema:
+           $ref: "#/definitions/Metrics" 
+        400:
+          description: "Bad request"
+          schema:
+            $ref: "#/definitions/Error"
+
+# Descriptions of common components
+definitions:
+  # Prometheus line exposition format
+  Metrics:
+    type: string
+    example: "http_requests_total{method='post',code='200'} 1027 1395066363000\nhttp_requests_total{method='post',code='400'}    3 1395066363000"
+  Process:
+    type: object
+    properties:
```

Comment:
lets add 'pid' as a property. We won't have it (yet) for processes connected via reversed pipe but we wouldn't have name or commandLine either.